pkg/controller/cincinnati/new.go

```diff
@@ -388,3 +450,24 @@ func newCMEnvVar(name, key, cmName string) corev1.EnvVar {
 		},
 	}
 }
+
+// checksumMap produces a checksum of a ConfigMap's Data attribute. The checksum
+// can be used to detect when the contents of a ConfigMap have changed.
+func checksumMap(m map[string]string) string {
+	keys := sort.StringSlice([]string{})
+	for k := range m {
+		keys = append(keys, k)
+	}
+	keys.Sort()
+
+	hash := sha256.New()
+	encoder := base64.NewEncoder(base64.StdEncoding, hash)
+
+	for _, k := range keys {
+		encoder.Write([]byte(k + ":"))
+		encoder.Write([]byte(m[k] + ":"))
```

Comment:
Any particular reason for appending  `:` at the end there?

Reply:
Generally I'm just trying to use a separator in a simple way that will definitely be distinct from the data. That way `{"foo": "bar"}` and `{"fo": "obar"}` won't serialize the same. Base64-encoding limits the character set, and then `:` was just an arbitrary choice of separator.

Reply:
Though now that I'm looking at it again, I don't think I accomplished what I intended. :) I may revise this...